core/store/src/trie/mod.rs

```diff
@@ -594,6 +594,16 @@ impl WrappedTrieChanges {
                 )),
                 "NotWritableToDisk changes must never be finalized."
             );
+            // Filtering trie keys.
+            // NOTE: If the trie key is not one of the account specific, it may cause key conflict
+            // when the node tracks multiple shards. See #2563.
+            match &change_with_trie_key.trie_key {
+                TrieKey::Account { .. }
+                | TrieKey::ContractCode { .. }
+                | TrieKey::AccessKey { .. }
+                | TrieKey::ContractData { .. } => (),
+                _ => continue,
+            };
```

Comment:
I am confused. Why are these keys skipped?

Reply:
If I understood correctly, by design of `StateChanges` - rather than a transaction log that can be an alternative to applying blocks, it's a user-facing log for RPC purposes only

Reply:
@mikhailOK that is correct. `StateChanges` are recorded specifically for RPC needs, and we only expose these types of changes.

Reply:
I see